src/main/java/com/google/api/codegen/transformer/py/PythonSurfaceNamer.java

```diff
@@ -453,4 +453,27 @@ public String getReleaseAnnotation(ReleaseLevel releaseLevel) {
         throw new IllegalStateException("Invalid development status");
     }
   }
+
+  @Override
+  public String getPrintSpec(String spec) {
+    // Escaers don't work here. They only map from characters to strings.
+    StringBuilder sb = new StringBuilder();
+    int cursor = 0;
+    while (true) {
+      int p = spec.indexOf('%', cursor);
+      if (p < 0) {
+        return sb.append(spec, cursor, spec.length()).toString();
+      }
+      sb.append(spec, cursor, p);
+
+      if (spec.startsWith("%%", p)) {
+        sb.append('%');
+      } else if (spec.startsWith("%s", p)) {
+        sb.append("{}");
+      } else {
+        throw new IllegalArgumentException(String.format("bad format verb: %s", spec));
```

Comment:
Something to think about for later: do we want to also support `%q` in the Go sense, so that it maps (for Python) to `sb.append("\"{}\"")`?

Reply:
:+1:   Agree on the P2 (hence the "later")